climpred/classes.py

```diff
@@ -201,8 +201,67 @@ def _construct_direct(cls, datasets):
         """
         obj = object.__new__(cls)
         obj._datasets = datasets
+        obj.kind = kind
         return obj
 
+    def _get_metric_comparison_dim(self, metric, comparison, dim):
+        if dim is None:
+            dim = 'init' if self.kind == 'hindcast' else ['init', 'member']
+
+        if self.kind == 'hindcast':
+            is_in_list(dim, ['member', 'init'], 'dim')
+        elif self.kind == 'perfect':
+            is_in_list(dim, ['member', 'init', ['init', 'member']], 'dim')
+
+        # get metric and comparison strings incorporating alias
+        metric = METRIC_ALIASES.get(metric, metric)
+        comparison = COMPARISON_ALIASES.get(comparison, comparison)
+
+        METRICS = HINDCAST_METRICS if self.kind == 'hindcast' else PM_METRICS
```

Comment:
I am on the ehh side for defining ALL_CAPS_CONSTANTS in a function

Reply:
What if we simply have
METRICS = {
    'hindcast': HINDCAST_METRICS,
    'pm': PM_METRICS
}